programs/lighthouse/src/utils.rs

```diff
@@ -32,15 +46,22 @@ pub fn unpack_coption_key(src: &[u8]) -> Result<Option<&Pubkey>> {
     }
 }
 
-pub fn unpack_coption_u64(src: &[u8]) -> Result<Option<u64>> {
-    let tag = &src[0..4];
-    let body = &src[4..12];
+pub fn unpack_coption_u64(src: &[u8], offset: usize) -> Result<Option<u64>> {
```

Comment:
This should be merged with `unpack_coption_key` and made into `unpack_coption<T: AnyBitPattern>` right?

Reply:
This is code I took from spl but yeah at this point good idea.

Reply:
Doesn't a u64 need to be 8 byte aligned to have its memory referenced using bytemuck? I've been told it might be dangerous for specific architectures.

Reply:
Pubkeys are [u8; 32] so its safe.

Reply:
Right, the other thing is you're returning a `u64` directly instead of a `&u64`. In that case, it might be easier to just avoid overcomplicating things and keep the current structure.